task3.py

```diff
@@ -0,0 +1,305 @@
+
+        timezone = pytz.FixedOffset(offset.total_seconds() // 60)
+        current_time = datetime.now(timezone).strftime('%Y-%m-%d %H:%M:%S')
+
+        row['current_time'] = current_time
+
+
+def create_folder_structure(data):
+    """
+    Creates folder structure based on data
+    :param data: A list of dictionaries representing the data used to create the folder structure
+    :return: A nested dictionary representing the folder structure
+    """
+    folder_structure = {}
+    for row in data:
+        dob_year = datetime.strptime(row['dob.date'], '%m/%d/%Y').year
+        decade = f'{dob_year // 10}0-th'
+        country = row['location.country']
+
+        if decade not in folder_structure:
+            folder_structure.setdefault(decade, {})
+        if country not in folder_structure[decade]:
+            folder_structure[decade].setdefault(country, [])
+
+        folder_structure[decade][country].append(row)
+
+    return folder_structure
+
+
+def calculate_max_age(data):
+    """
+    Calculates the maximum age from a list of data
+    :param data: A list of dictionaries representing the data containing date of birth information
+    :return: The maximum age calculated from the data
+    """
+    dob_dates = [datetime.strptime(row['dob.date'], '%m/%d/%Y') for row in data]
+    return max((datetime.today() - date).days // 365 for date in dob_dates)
+
+
+def calculate_avg_registered_years(data):
+    """
+    Calculates the average number of registered years from a list of data
+    :param data: A list of dictionaries representing the data containing registration date information
+    :return: The average number of registered years
+    """
+    reg_dates = [datetime.strptime(row['registered.date'], '%m-%d-%Y, %H:%M:%S') for row in data]
+    return sum((datetime.today() - date).days // 365 for date in reg_dates) / len(data)
+
+
+def calculate_popular_id(data):
+    """
+    Calculates the most popular id from a list of data
+    :param data: A list of dictionaries representing the data containing id information
+    :return: The most popular id
+    """
+    name_counts = Counter(row['id.name'] for row in data)
+    return name_counts.most_common(1)[0][0]
+
+
+def create_folders(destination_folder, folder_structure):
+    """
+    Creates folders based on a nested dictionary representing folder structure
+    :param destination_folder: The path to the destination folder where the folders will be created
+    :param folder_structure: A nested dictionary representing the folder structure
+    :return: The updated folder structure dictionary
+    """
+    for decade, countries in folder_structure.items():
+        decade_folder = os.path.join(destination_folder, decade)
+        os.makedirs(decade_folder, exist_ok=True)
+
+        for country in countries:
+            country_folder = os.path.join(decade_folder, country)
+            os.makedirs(country_folder, exist_ok=True)
+
+    return folder_structure
```

Comment:
the second makedirs should create the whole hierarchy.
so you can skip the first one.

you can pass to **path.join** as many arguments as you need

if you have some nested code. you can write return without blank line after the nested code. because it already will be visually separated.
```suggestion
    for decade, countries in folder_structure.items():
        for country in countries:
            country_folder = os.path.join(destination_folder, decade, country)
            os.makedirs(country_folder, exist_ok=True)
    return folder_structure
```